src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -638,7 +638,7 @@ public void createSnapshotWithProvidedIds(
     public void grantReadAccessToSnapshot(Snapshot snapshot, Collection<String> policies) throws InterruptedException {
         // TODO: When we support multiple datasets per snapshot, this will need to be reworked
         grantReadAccessWorker(
-            bigQueryProjectForDataset(snapshot.getFirstSnapshotSource().getDataset()),
+            bigQueryProjectForSnapshot(snapshot),
```

Comment:
Are we sure that this is what we want?  I thought we wanted to give read access to the underlying dataset?  I guess this is a recent change.  Did you manually check this?

Reply:
Yes, I changed this line back to what is currently on [develop](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java#L605).

This call to `grantReadAccessWorker` calls `datasetBigQueryProject.addDatasetAcls`, which tries to get the BigQuery dataset for the snapshot and update its ACLs. If the snapshot is in a different project than the dataset it looks like this should remain `bigQueryProjectForSnapshot(snapshot)` (otherwise it throws a null pointer exception).

Reply:
Understood, thanks for clarifying!  I just wanted to make extra sure